frontend/src/scenes/insights/InsightNav/insightNavLogic.tsx

```diff
@@ -297,26 +328,32 @@ const mergeCachedProperties = (query: InsightQueryNode, cache: QueryPropertyCach
     if (isInsightQueryWithSeries(mergedQuery)) {
         if (cache.series) {
             if (isLifecycleQuery(mergedQuery)) {
-                mergedQuery.series = cache.series.slice(0, 1)
+                mergedQuery.series = cleanSeriesMath(cache.series.slice(0, 1), MathAvailability.None)
             } else {
-                mergedQuery.series = cache.series
+                const mathAvailability = isTrendsQuery(mergedQuery)
+                    ? MathAvailability.All
+                    : isStickinessQuery(mergedQuery)
+                    ? MathAvailability.ActorsOnly
+                    : MathAvailability.None
+                mergedQuery.series = cleanSeriesMath(cache.series, mathAvailability)
             }
-        } else if (cache.retentionFilter?.targetEntity || cache.retentionFilter?.returningEntity) {
-            mergedQuery.series = [
-                ...(cache.retentionFilter.targetEntity
-                    ? [cache.retentionFilter.targetEntity as EventsNode | ActionsNode]
-                    : []),
-                ...(cache.retentionFilter.returningEntity
-                    ? [cache.retentionFilter.returningEntity as EventsNode | ActionsNode]
-                    : []),
-            ]
         }
+        // else if (cache.retentionFilter?.targetEntity || cache.retentionFilter?.returningEntity) {
+        //     mergedQuery.series = [
+        //         ...(cache.retentionFilter.targetEntity
+        //             ? [cache.retentionFilter.targetEntity as EventsNode | ActionsNode]
+        //             : []),
+        //         ...(cache.retentionFilter.returningEntity
+        //             ? [cache.retentionFilter.returningEntity as EventsNode | ActionsNode]
+        //             : []),
+        //     ]
+        // }
```

Comment:
🤔

Reply:
Should stay and will be added back with when the retention entities are converted to the new format. It's commented out for now to circumvent this issue:

> Works around a bug (https://github.com/PostHog/posthog/issues/20108) with retention insights by simply not setting the retention entities from series.

Downside is that retention entities don't get set automatically when switching between insight types, but don't think that's too big of a nuisance at the moment.